### DISCUSSION TITLE
Debian packaging

### DISCUSSION THREAD

**Discussion by hsbasu** [amit9838/mousam#68] [Show and tell]
I am building continuous debian packages for `mousam` at [Launchpad](https://code.launchpad.net/~mamolinux/+recipe/mousam). If anyone is interested, they can use the following commands to install or upgrade `mousam`
```
sudo add-apt-repository ppa:mamolinux/gui-apps
sudo apt update
sudo apt install mousam
```
This PPA provides latest stable releases when a new version is released. These are built from the [mamolinux/stable](https://github.com/mamolinux/mousam/tree/mamolinux/stable) branch.

For daily builds, use 
```
sudo add-apt-repository ppa:mamolinux/daily-builds
sudo apt update
sudo apt install mousam
```
**Warning:** This ppa provides latest updates as soon as they are added to the master branch. If you are using cinnamon desktop, all cinnamon related packages might get updated to an unstable version.

**Comment by SJaafar** [comment]
When adding the repository in Debian, I get these lines:

> Ign:6 https://ppa.launchpadcontent.net/mamolinux/gui-apps/ubuntu bookworm InRelease
> Err:7 https://ppa.launchpadcontent.net/mamolinux/gui-apps/ubuntu bookworm Release
>   404  Not Found [IP: 185.125.190.80 443]
> Reading package lists... Done
> E: The repository 'https://ppa.launchpadcontent.net/mamolinux/gui-apps/ubuntu bookworm Release' does not have a Release file.
> N: Updating from such a repository can't be done securely, and is therefore disabled by default.
> N: See apt-secure(8) manpage for repository creation and user configuration details

**Comment by hsbasu** [comment]
This is happening because you did not follow the instructions on Launchpad. LP only publishes for ubuntu versions like `mantic`, `noble` etc. Add the repo using the following commands.
```
sudo add-apt-repository ppa:mamolinux/gui-apps
sudo apt update
```
This is also mentioned on [LP](https://launchpad.net/~mamolinux/+archive/ubuntu/gui-apps).

Or, first get the signing key using 
```
wget --no-check-certificate -qO - https://raw.githubusercontent.com/mamolinux/mamolinux-archive/master/data/mamolinux-keyring-2021-archive.gpg | gpg --dearmour | sudo tee /etc/apt/trusted.gpg.d/mamolinux-keyring-2021-archive.gpg > /dev/null
```
Then use Ubuntu's version name like
```
deb https://ppa.launchpadcontent.net/mamolinux/gui-apps/ubuntu noble main 
# deb-src https://ppa.launchpadcontent.net/mamolinux/gui-apps/ubuntu noble main
```

For newer Ubuntu versions (>=24.04), create a file `gui-apps.sources` in `/etc/apt/sources.list.d/` and put the following lines
```
Types: deb
URIs: https://ppa.launchpadcontent.net/mamolinux/gui-apps/ubuntu
Suites: noble
Components: main
Signed-By: /etc/apt/trusted.gpg.d/mamolinux-keyring-2021-archive.gpg
```

**Comment by Rohlik** [comment]
Getting this error on my Ubuntu Jammy (22.04):
```
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: https://prod-cdn.packages.k8s.io/repositories/isv:/kubernetes:/core:/stable:/v1.29/deb  InRelease: The following signatures were invalid: EXPKEYSIG 234654DA9A296436 isv:kubernetes OBS Project <isv:kubernetes@build.opensuse.org>
W: Failed to fetch https://pkgs.k8s.io/core:/stable:/v1.29/deb/InRelease  The following signatures were invalid: EXPKEYSIG 234654DA9A296436 isv:kubernetes OBS Project <isv:kubernetes@build.opensuse.org>
W: Some index files failed to download. They have been ignored, or old ones used instead.
```

**Comment by Bikkjevov** [comment]
Changing location once worked. On the second attempt, the application crashed. dmesg shows:
traps: mousam[144981] general protection fault ip:70f57e16edc4 sp:7ffc980cbd30 error:0 in libgtk-4.so.1.1400.5[16edc4,70f57e0b0000+49b000]

Linux Mint 22.3, Cinnamon desktop
